crates/metastore/src/storage/lease.rs

```diff
@@ -250,6 +246,13 @@ impl LeaseRenewer {
                 });
             }
 
+            if expires_at - LEASE_DURATION + (LEASE_RENEW_INTERVAL / 2) > now {
+                // Don't acquire a new lease. Let the process use the existing
+                // lease. This should help in reducing the number of writes to
+                // the lease object.
+                return Ok(lease.generation);
+            }
+
```

Comment:
@scsmithr @tychoish ^

This should really solve everything...

Reply:
I just want to clarify things:
- is there a way that the same catalog could be modified or accessed by two tasks/threads within the same process? My concern is that this lease could be (unintentionally) shared between two orthogonal processes. 
- what is the case when the lease is (actually) renewed, and in what cases are leases released? Does this change any of those points?

Reply:
> is there a way that the same catalog could be modified or accessed by two tasks/threads within the same process? My concern is that this lease could be (unintentionally) shared between two orthogonal processes.

So, I just took a better look and `process_id` refers to the service and not session. So, it's most definitely possible that two threads within the same process modify the same catalog! I should undo this change.

> what is the case when the lease is (actually) renewed, and in what cases are leases released? Does this change any of those points?

Lease is renewed after the given interval. A lease is released once the mutation is over (or automatically if it expires).

Reply:
> So, it's most definitely possible that two threads within the same process modify the same catalog! I should undo this change.

Maybe. If this is an existing flaw, and this change doesn't increase the likelihood of a collision, then it's probably safe to commit it. We could also change to something that isn't the `process_id`.